concordium-sdk/src/main/java/com/concordium/sdk/Client.java

```diff
@@ -330,6 +333,23 @@ public ImmutableList<BakerId> getBakerList(Hash blockHash) throws BlockNotFoundE
         return ImmutableList.copyOf(bakerIdArray);
     }
 
+    /**
+     * Get the status of a given baker pool or passive delegation at the given block.
+     *
+     * @param blockHash {@link Hash} of the block Pool Status is to be retrieved at.
+     * @param bakerId   {@link Optional<BakerId>} of the Baker of which pool is to be retrieved.
+     * @return Parsed {@link PoolStatus}.
+     */
+    public PoolStatus getPoolStatus(final Hash blockHash, final Optional<BakerId> bakerId) {
+        val res = server().getPoolStatus(ConcordiumP2PRpc.GetPoolStatusRequest.newBuilder()
+                .setBlockHash(blockHash.asHex())
+                .setPassiveDelegation(!bakerId.isPresent())
+                .setBakerId(bakerId.orElse(BakerId.from(0)).getId().getIndex().getValue())
+                .build());
+
```

Comment:
Can the result be `null` ? If so please act appropriately. 

I suggest something like `AccountNotFoundException`.

Reply:
done